### PR TITLE
add retrieveMe() method in the User class.

### PR DESCRIPTION
**Pull request by jchen293**
Add **retrieveMe()** helper method in the User class, the users can call this method to grab their profile. E2E local test was done for this PR.

### CODE REVIEW

File: user.go [hunk 1]
```diff
@@ -100,3 +100,16 @@ func (c *Client) DeleteUser(userID string) error {
 func (c *Client) DeleteUserWithContext(ctx context.Context, userID string) error {
 	return c.del(ctx, "users/"+userID)
 }
+
+// RetrieveMe retrieve the current user profile.
```
Comment (Justintime50) [inline user.go:104]:
```suggestion
// RetrieveMe retrieves the current user.
```

File: user.go [hunk 2]
```diff
@@ -100,3 +100,16 @@ func (c *Client) DeleteUser(userID string) error {
...
+func (c *Client) RetrieveMe() (out *User, err error) {
+	c.get(nil, "users/", &out)
```
Comment (Justintime50) [inline user.go:106]:
```suggestion
	c.get(nil, "users", &out)
```

File: user.go [hunk 3]
```diff
@@ -100,3 +100,16 @@ func (c *Client) DeleteUser(userID string) error {
...
+	return
+}
+
+// RetrieveMeWithContext performs the same operation as RetrieveMe, but allows
+// specifying a context that can interrupt the request.
+func (c *Client) RetrieveMeWithContext(ctx context.Context) (out *User, err error) {
+	c.get(ctx, "users/", &out)
```
Comment (Justintime50) [inline user.go:113]:
```suggestion
	c.get(ctx, "users", &out)
```

**Review by nwithan8 — APPROVED** [review]

File: user.go (hunk 1 above)
Comment (Justintime50) [reply user.go]: This wasn't completely resolved - these aren't really "profiles", but I won't block on that either.

**Review by Justintime50 — APPROVED** [review]

File: user.go (hunk 1 above)
Comment (jchen293) [reply user.go]: My bad I will address this in the next PR.

**Pull request merged**